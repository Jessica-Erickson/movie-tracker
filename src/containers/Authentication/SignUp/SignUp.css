* {
  font-family: 'Roboto Slab', serif;

}

.sign-up-form {
  background-color: #CACAD1;
  border-radius: 10px;
  border-color:  #37424E;
  border-width: 2px;
  box-shadow: 10px 9px 20px 0px;
  display: flex;
  flex-direction: column;
  padding-left: 3.5%;
  padding-bottom: 10px;
  padding-top: 10px;
  position: absolute;
  width: 250px;
  z-index: 10;
}

.name {
  border-radius: 5px;
  border-color:  #37424E;
  font-family: 'Roboto Slab', serif;
  height: 1.25rem;
  margin-top: 10px;
  padding-left: 5px;
  width: 75%;
}

.submit {
  margin-top: 100px;
}

.sign-in {
  color: #990000;
  font-size: 1.25rem;
  margin-left: 3.75rem;
  margin-top: 15px;
  text-decoration: none;
}