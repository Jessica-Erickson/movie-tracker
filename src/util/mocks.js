import { key } from './key.js'

export const fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(mock.mockResults)
      }))

export const mockResults = { 
  results: [
    { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }
  ] 
}

export const mockMovie = { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }

export const mockState = { 
    movies: [
      { 
        poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
        title: 'Venom',
        vote_average: 5 ,
        overview: 'This movie is RAD!',
        release_date: '2018-9-11',
      }
    ],
    filter: 'favorites' 
  };

export const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;

export const userResponse = {
  data: {
    id: 2,
    name: 'Jessica',
    password: 'passw0rd',
    email: 'jerick999@gmail.com',
  },
  message: 'Retrieved ONE User',
  status: 'success',
}

export const user = {
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
}

export const options = {
  method: 'POST',
  body: JSON.stringify({
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}

export const newUserResponse = {
  id: 4,
  message: "New user created",
  status: "success",
}

export const newUser = {
  name: 'Tim', 
  password: 'm30w',
  email: 'tag.garrity@gmail.com', 
}

export const newOptions = {
  method: 'POST',
  body: JSON.stringify({
  name: 'Tim',
  password: 'm30w',
  email: 'tag.garrity@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}