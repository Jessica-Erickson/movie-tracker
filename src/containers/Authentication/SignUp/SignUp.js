import React, { Component }  from 'react';
import * as API from '../../../util/helper'

export class SignUp extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      error: '',
    }

  }

  updateValue = (e) => {
    const { value, name } = e.target;
    this.setState({
      [name]: value
    })
  }

  submitNewUser = async (e) => {
    const { name , email , password } = this.state;
    const user = { name , email , password };
    e.preventDefault();
    try {
      await API.addUser(user)
      this.setState({ error: '' });
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const { name, email, password , error } = this.state
    return (
      <form 
      onSubmit={this.submitNewUser}>
        { error !== '' && <h2>{error}</h2> }
        <input
          className='name' 
          onChange={this.updateValue}
          name="name" 
          value={name} 
          placeholder="Enter your name"/>
        <input 
          className='email' 
          onChange={this.updateValue}
          name="email" 
          value={email} 
          placeholder="Enter your email"/>
        <input 
          className='password' 
          onChange={this.updateValue}
          name="password" 
          value={password} 
          type="password" 
          placeholder="Enter your password"/>
        <input 
          type='submit'  
          value='Sign Up'/>
      </form>
    )
  }



}