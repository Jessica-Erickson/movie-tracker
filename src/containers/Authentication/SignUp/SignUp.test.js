import SignUp from './SignUp.js';
import * as mock from './mocks.js';
import { shallow } from 'enzyme';


describe('SignUp', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow( <SignUp/> )
  }) 
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})