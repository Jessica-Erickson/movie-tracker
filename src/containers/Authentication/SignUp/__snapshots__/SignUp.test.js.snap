// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2 />
  <input
    className="name"
    name="name"
    onChange={[Function]}
    placeholder="Enter your name"
    value=""
  />
  <input
    className="email"
    name="email"
    onChange={[Function]}
    placeholder="Enter your email"
    value=""
  />
  <input
    className="password"
    name="password"
    onChange={[Function]}
    placeholder="Enter your password"
    type="password"
    value=""
  />
  <input
    type="submit"
    value="Sign Up"
  />
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/login"
  >
    Sign In
  </NavLink>
</form>
`;
